### PR TITLE
Support trivial manifest list comparations

### PR DESCRIPTION
**Pull request by rporres**
[Docker v2 schema 2 manifest spec](https://docs.docker.com/registry/spec/manifest-v2-2/) supports multiarch images via manifest lists. This patch makes sures that we compare v2 schema 2 images properly to avoid the `__eq__` method to fail when we compare single arch and multiarch images.

A reproducer to the behaviour being fixed

```
$ docker pull jaegertracing/jaeger-agent:1
$ docker tag jaegertracing/jaeger-agent:1 quay.io/rporresm/jaegertracing-jaeger-agent:1
$ docker push quay.io/rporresm/jaegertracing-jaeger-agent:1

$ cat reproducer.py
from sretoolbox.container.image import Image
multi_image = Image("docker://docker.io/jaegertracing/jaeger-agent:1")
simple_image = Image("docker://quay.io/rporresm/jaegertracing-jaeger-agent:1")
print(multi_image == simple_image)

$ python reproducer.py
Traceback (most recent call last):
  File "/Users/rporresm/git/github.com/rporres/sretoolbox/simple-reproduce.py", line 5, in <module>
    print(multi_image == simple_image)
  File "/Users/rporresm/git/github.com/rporres/sretoolbox/sretoolbox/container/image.py", line 477, in __eq__
    if manifest[layers_key] == other_manifest[layers_key]:
KeyError: 'layers'
```

`multi_image` manifest
```
{'mediaType': 'application/vnd.docker.distribution.manifest.list.v2+json',
 'schemaVersion': 2,
 'manifests': [{'mediaType': 'application/vnd.docker.distribution.manifest.v2+json',
                'digest': 'sha256:49dc93ebebcb857a999c2f7e1ed35dffd92c69371803e43d09a1be136c14b7df',
                'size': 1159,
                'platform': {'architecture': 'amd64', 'os': 'linux'}},
               {'mediaType': 'application/vnd.docker.distribution.manifest.v2+json',
                'digest': 'sha256:e27e6af5aa98f19a2afae3cc5f133b9111984502c0c9f50a0393e2edac4f2a17',
                'size': 1159,
                'platform': {'architecture': 's390x', 'os': 'linux'}},
               {'mediaType': 'application/vnd.docker.distribution.manifest.v2+json',
                'digest': 'sha256:26d84e86ce9b984355e923f679a52286b3880bbcec4581c24d5a95e1bb9d274f',
                'size': 1159,
                'platform': {'architecture': 'ppc64le', 'os': 'linux'}},
               {'mediaType': 'application/vnd.docker.distribution.manifest.v2+json',
                'digest': 'sha256:feaca2b9c9c93c4f8f8dfebb3a2a34232ee282bd05846e233090fd797687cadd',
                'size': 1159,
                'platform': {'architecture': 'arm64', 'os': 'linux'}}]}
{'schemaVersion': 2,
```

`simple_image` manifest
```
{'schemaVersion': 2,
 'mediaType': 'application/vnd.docker.distribution.manifest.v2+json',
 'config': {'mediaType': 'application/vnd.docker.container.image.v1+json',
            'size': 2384,
            'digest': 'sha256:c286c20181c8dd781f88b92fd7b604a5da19ace9394bd7c8ce2f712d9891e7fc'},
 'layers': [{'mediaType': 'application/vnd.docker.image.rootfs.diff.tar.gzip',
             'size': 2818370,
             'digest': 'sha256:8663204ce13b2961da55026a2034abb9e5afaaccf6a9cfb44ad71406dcd07c7b'},
            {'mediaType': 'application/vnd.docker.image.rootfs.diff.tar.gzip',
             'size': 117402,
             'digest': 'sha256:d20664ba4ab4e000e4c29687a774df1b56982774a4de1067120a463001e1f0a6'},
            {'mediaType': 'application/vnd.docker.image.rootfs.diff.tar.gzip',
             'size': 18735,
             'digest': 'sha256:855e86ef411856bc3903354d76e919e34c38f9602a067d9b022d0ae4f7baa6d1'},
            {'mediaType': 'application/vnd.docker.image.rootfs.diff.tar.gzip',
             'size': 12155076,
             'digest': 'sha256:6c8f501f526b9139a396fb88eff8ee23baaf0b9df0e2506b07ba0a137e07a8f7'}]}
```

There's no test of the current behaviour and no test has been provided due the complexity of the manifest fetching. I could easily mock the `_get_manifest` method, but then the test is not extremely useful. I could work on it if deemed necessary, but I want to get some feedback of the general approach.

The proposed behaviour looks correct to me: if I'm trying to compare an image that returns a single manifest with one that has many manifests, it should return false. On the other hand this looks short sighted as the manifest on one side can be part of the manifest lists. Most likely a more advanced and configurable `eq` method should be written as this looks not enough. But it is beyond the scope of this patch.

Signed-off-by: Rafa Porres Molina <rporresm@redhat.com>

### CODE REVIEW

**Comment by rporres** [comment]
cc @apahim

**Review by chassing — APPROVED** [review]
LGTM

File: sretoolbox/container/image.py [hunk 1]
```diff
@@ -472,7 +473,21 @@ def __eq__(self, other):
         if manifest_version == 1:
             layers_key = 'fsLayers'
         else:
-            layers_key = 'layers'
+            manifest_media_type = manifest['mediaType']
+            other_manifest_media_type = other_manifest['mediaType']
+
+            if manifest_media_type != other_manifest_media_type:
```
Comment (apahim) [inline sretoolbox/container/image.py:479]: agreed... __eq__ will aways have a meaning and in this case it has the correct meaning: images are not the same.

Thinking about the extended scope, when a simple image can be in the manifest on the multi-image, the interface I would expect is something like `simple_image in multi_image.manifest`, but yeah, that's a different topic :)

**Review by apahim — APPROVED** [review]
lgtm

File: sretoolbox/container/image.py (hunk 1 above)
Comment (rporres) [reply sretoolbox/container/image.py:479]: That interface could work for me as well ;-)

**Pull request merged**